torch/utils/_pytree.py

```diff
@@ -52,7 +51,7 @@ class NodeDef(NamedTuple):
     flatten_fn: FlattenFunc
     unflatten_fn: UnflattenFunc
 
-SUPPORTED_NODES: Dict[Type[Any], NodeDef] = {}
+SUPPORTED_NODES: Dict[int, NodeDef] = {}
```

Comment:
meh but ok (but only because we typically don't expect types to get deallocated). I feel it shouldn't be that difficult to get Dynamo to support dict with Type input though...

Reply:
The problem was how to materialize the sources for each type (+ `namedtuple` function) key in this dictionary. Would iterating through the type's `__qualname__` for that be a preferable solution?

Reply:
This is for guards, right? I feel like we want something like https://github.com/pytorch/pytorch/pull/108528/ where we don't bother guarding on anything in SUPPORTED_NODES, and just observe when the dict is modified.